lock_api/src/rwlock.rs

```diff
@@ -56,6 +56,29 @@ pub unsafe trait RawRwLock {
 
     /// Releases an exclusive lock.
     fn unlock_exclusive(&self);
+
+    /// Checks if this `RwLock` is currently locked in any way.
+    fn is_locked(&self) -> bool {
+        self.is_locked_shared() || self.is_locked_exclusive()
```

Comment:
I think we should only expose a single `is_locked` function here. I don't think there is much value in trying to differentiate between readers and writers.